test/e2e/versioned_explicit_reference_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	uitest "github.com/cppforlife/go-cli-ui/ui/test"
+)
+
+func TestVersionedExplicitReference(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+
+	yaml1 := `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-1
+  annotations:
+    kapp.k14s.io/versioned: ""
+data:
+  foo: bar
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-2
+  annotations:
+    kapp.k14s.io/versioned-explicit-ref: '{ "references": [ { "namespace": "kapp-test", "kind" : "ConfigMap", "name": "config-1"} ] }'
+data:
+  foo: bar
+`
+
+	yaml2 := `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-1
+  annotations:
+    kapp.k14s.io/versioned: ""
+data:
+  foo: bar1
```

Comment:
Should we use a totally different value instead of `bar1`, like `blahblahblah`, which is easy to identify? Right now this is the only difference in yaml1 and yaml2, so it's a bit difficult to spot the difference. Or maybe we can use replace, as it would be used only in a few places.

Reply:
Not sure about a replace, but yep a significantly different value will certainly help. Will do!